state-sync/state-sync-driver/src/notification_handlers.rs

```diff
@@ -263,29 +263,40 @@ impl ConsensusNotificationHandler {
     pub async fn initialize_sync_target_request(
         &mut self,
         sync_target_notification: ConsensusSyncTargetNotification,
+        latest_pre_committed_version: Version,
         latest_synced_ledger_info: LedgerInfoWithSignatures,
     ) -> Result<(), Error> {
-        // Get the latest committed version and the target sync version
+        // Get the target sync version and latest committed version
         let sync_target_version = sync_target_notification
             .get_target()
             .ledger_info()
             .version();
         let latest_committed_version = latest_synced_ledger_info.ledger_info().version();
 
         // If the target version is old, return an error to consensus (something is wrong!)
-        if sync_target_version < latest_committed_version {
+        if sync_target_version < latest_committed_version
+            || sync_target_version < latest_pre_committed_version
+        {
             let error = Err(Error::OldSyncRequest(
                 sync_target_version,
+                latest_pre_committed_version,
                 latest_committed_version,
             ));
             self.respond_to_sync_target_notification(sync_target_notification, error.clone())?;
             return error;
         }
 
         // If we're now at the target, return successfully
-        if sync_target_version == latest_committed_version {
-            info!(LogSchema::new(LogEntry::NotificationHandler)
-                .message("We're already at the requested sync target version! Returning early"));
+        if sync_target_version == latest_committed_version
+            || sync_target_version == latest_pre_committed_version
```

Comment:
we can't return Ok if it's equal to the pre commit version, we need to somehow write the ledger info down or just simply return Err